lib/epub_extact.ex

```diff
@@ -0,0 +1,34 @@
+defmodule EpubExtract do
+  alias Floki
+
+  def unzip_and_parse_epub(epub_path) do
+    temp_dir = create_temp_dir()
+
+    {:ok, _} = unzip_epub(epub_path, temp_dir)
+
+    extracted_content = read_temp_dir(temp_dir)
+    parsed_segments = parse_with_floki(extracted_content)
+    :ok = File.rm_rf!(temp_dir)
+
+    parsed_segments
+  end
+
+  defp unzip_epub(epub_path, extract_path) do
+    epub_file = File.read!(epub_path)
+    :zip.unzip(epub_file, [{:cwd, extract_path}])
+  end
+
+  defp create_temp_dir do
+    {:ok, temp_dir} = File.Temp.dir!("epub_extraction")
+    temp_dir
+  end
+
+  defp read_temp_dir(temp_dir) do
+    File.read!(temp_dir)
+  end
+
+  defp parse_with_floki(content) do
```

Comment:
maybe you need to expand this function a bit more

Reply:
like?

Reply:
Well for instance you still need to get the output where you pass in one single XHTML file and then parse and split paragraph by paragraph?